app/client/ui-modules/guest-landing-page/SearchResultPage.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useEffect, useMemo, useState } from "react";
+import { Button } from "../theming-shadcn/Button";
+import { PropertyCard } from "./components/PropertyCard";
+import { Property } from "/app/client/library-modules/domain-models/property/Property";
+import { getPropertyWithListingDataUseCase } from "../../library-modules/use-cases/property-listing/GetPropertyWithListingDataUseCase";
+import { PropertyWithListingData } from "../../library-modules/use-cases/property-listing/models/PropertyWithListingData";
+import { searchProperties } from "../../library-modules/domain-models/property/repositories/property-repository";
+
+import { Input } from "../theming-shadcn/Input";
+import { useLocation, useNavigate } from "react-router";
+
+// load up 9 properties at a time
+const PAGE_SIZE = 9;
+
+// Utility to get the "q" param from the URL
+function getQueryParam(search: string): string {
+    const params = new URLSearchParams(search);
+    return params.get("q") ?? "";
+}
+
+export function GuestSearchResultsPage() {
```

Comment:
this is a suggestion but i think this file/func is doing too much.It's not just showing the results, it's also handling all the work to get the data, track loading, and deal with errors.

maybe consider moving all that data-fetching logic into its own separate function or like a hook